app/models/spell.rb

```diff
@@ -25,18 +52,35 @@ class Spell < ApplicationRecord
 
   def validate_spell_requirements
     self.spell_requirements.values.each do |req_value|
-      if req_value != true && req_value != false    
+      if req_value != true && req_value != false
         errors.add("Spell requirements", "All values for the spell requirements must be true or false.")
       end
-    end    
+    end
   end
 
   def validate_class_requirements
     self.class_requirements.values.each do |req_value|
       if req_value.class != Integer && req_value != nil
         errors.add("Class requirements", "All values for the class requirements must be an integer or nil.")
       end
-    end    
+    end
+  end
+
+  def list_spell_requirements
+    display = ""
+    self.spell_requirements.each do |requirement, value|
+      if value
+        display << requirement[0]
+      end
+    end
+
+    if display.include?("f")
```

Comment:
This block is to sort the components for display. May not be needed if we ever get around to an Ember front end, but it currently takes the `spell_requirements` hash and creates a string "V, S, M, F"

Reply:
ah got you. Would it be easier to sort them on save/creation and then not really worry about the display? Is that possible?

Reply:
I don't think there is a way to do it at creation because the sort methods convert it to an array. In the seed file I define the hash in the VSMF order, but that is irrelevant when calling a Spell's requirements.

Reply:
I vote we leave it like this for now and maybe leave a comment that we should be able to extract this logic to somewhere else in the future